Fix: Make reasoning_effort optional

Chesars opened the pull request:
## Summary
  Users with unverified OpenAI organizations encounter 400 errors because:
  1. The SDK automatically sets `reasoning_effort='high'` by default
  2. The SDK includes `reasoning.summary` in API calls, which requires verified organizations

  This PR fixes both issues by making `reasoning_effort` opt-in and removing the automatic `summary` parameter.

  ## Details
  OpenAI's Responses API allows `reasoning.effort` for all organizations, but `reasoning.summary` requires verification. 
  **Changes:**
  1. Removed auto-assignment of `reasoning_effort='high'` in `llm.py`
  2. Removed `summary` parameter from reasoning object in `responses_options.py`
  3. Made reasoning conditional - only sent when explicitly set by user

  ## Code
  - `openhands-sdk/openhands/sdk/llm/llm.py:293-299`: Removed default reasoning_effort assignment
  - `openhands-sdk/openhands/sdk/llm/options/responses_options.py:43-44`: Removed summary from reasoning object

  Fixes #987

Chesars committed:
Fix: Make reasoning_effort optional to avoid API errors with unverified orgs

Remove automatic default of reasoning_effort='high' and only include
reasoning parameter in API calls when explicitly set by the user. This
fixes 400 errors for users with unverified OpenAI organizations that
don't have access to reasoning models.

Changes:
- Remove auto-assignment of reasoning_effort='high' in llm.py
- Make reasoning parameter conditional in responses_options.py

Fixes #987

Chesars committed:
Merge branch 'main' into fix-reasoning-effort-optional

Chesars committed:
Update tests to reflect optional reasoning_effort behavior

Modify tests to expect reasoning_effort=None by default instead of
automatically defaulting to 'high'. This aligns tests with the fix
for issue #987.

Chesars committed:
Merge branch 'fix-reasoning-effort-optional' of https://github.com/Chesars/software-agent-sdk into fix-reasoning-effort-optional

Chesars committed:
Fix test to explicitly set reasoning_effort

Chesars committed:
Remove summary from reasoning to support unverified orgs

The reasoning.summary parameter requires verified OpenAI organizations,
while reasoning.effort works for all orgs. Remove the automatic inclusion
of summary to allow unverified organizations to use reasoning_effort.

This prevents 400 errors: "Your organization must be verified to generate
reasoning summaries."

Chesars committed:
Merge branch 'main' into fix-reasoning-effort-optional

enyst commented on openhands-sdk/openhands/sdk/llm/options/responses_options.py:
```diff
@@ -39,8 +39,8 @@ def select_responses_options(
-    # Request plaintext reasoning summary
-    effort = llm.reasoning_effort or "high"
-    out["reasoning"] = {"effort": effort, "summary": "detailed"}
+    # Include reasoning effort only if explicitly set (without summary to avoid requiring verified org)
```
I think people need summary too 🤔

You’re right it shouldn’t be default, since it requires identification, but what do you think about adding it as LLM instance field? 🤔 With a default of None, so we can skip it here

Chesars committed:
Add optional reasoning_summary field for verified orgs

Add reasoning_summary as an optional LLM field that allows users with
verified OpenAI organizations to explicitly request reasoning summaries.

Changes:
- Add reasoning_summary field to LLM class (default: None)
- Conditionally include summary in reasoning object when set
- Add test for reasoning_summary functionality

This maintains backward compatibility while enabling verified orgs to
use reasoning summaries when needed.

Chesars committed:
Merge branch 'fix-reasoning-effort-optional' of https://github.com/Chesars/software-agent-sdk into fix-reasoning-effort-optional

Chesars committed:
Merge branch 'main' into fix-reasoning-effort-optional

Chesars committed:
Fix line length for ruff linter

Chesars committed:
Merge branch 'fix-reasoning-effort-optional' of https://github.com/Chesars/software-agent-sdk into fix-reasoning-effort-optional

enyst reviewed (commented):
Thank you for looking into this! A few more notes about `reasoning_effort`:

Technically, setting it `None` means today different things:
- it gets set to `medium`
- except for gpt-5-pro, which gets set to `high`

https://platform.openai.com/docs/api-reference/responses/create#responses_create-reasoning-effort

Moreover, Gemini-2.5-Pro I think, could not be `None`. At the moment, I find in their API docs that "thinking can't be turned off":

https://docs.cloud.google.com/vertex-ai/generative-ai/docs/thinking#budget

I know the code looked weird as it was, but there were some weird reasons in the past, too 😅 

In addition, though, we did want "high" at the beginning, because we assumed it was going to be better if users get the best experience. Since then, as you see with Gemini, actually the best experience might have been with _lower thinking budget..._ 🤔 And today I hear a lot of talk about very fast models...

IMHO you're right that the most expected behavior, from client developers, is to set this to `None`, if we can (if we don't get errors from LLM APIs), and let the user specify their wanted value. So far, we have run benchmark evaluations with "high", so that means that we need to know and set it in evaluations explicitly.

Chesars committed:
Merge branch 'main' into fix-reasoning-effort-optional

Chesars committed:
Merge branch 'main' into fix-reasoning-effort-optional

enyst commented on openhands-sdk/openhands/sdk/llm/llm.py:
```diff
@@ -214,6 +214,12 @@ class LLM(BaseModel, RetryMixin, NonNativeToolCallingMixin):
         "Can apply to all reasoning models.",
     )
+    reasoning_summary: Literal["auto", "concise", "detailed"] | None = Field(
+        default=None,
```
I'm not sure if 'auto' means `None` for unverified?

Chesars replied:
Great report @enyst 👏. Glad to hear your opinion and those very well-founded arguments.

> I'm not sure if 'auto' means `None` for unverified?

Not at all. `auto` switches between `detailed` or `concise` depending on the model. But with `default="auto"`, unverified orgs would still receive **400 errors** - any summary value requires verification.

 What @xingyaoww mention could be also, to set up the "best" parameters `reasoning.effort=high` `reasoning.summary=detailed` (higher costs and break for unverified orgs) and document how to switch the
  configuration in the SDK. But as @enyst mention and I think as well, that a default of `None` could work for mostly users and also OpenAi also mentions "reasoning summaries are opt-in and not included by default". Both approaches would work great with some documentation, I lean towards `default=None` for reasoning.summary for making the SDK work out-of-the-box for users. WDYT?

Chesars committed:
Merge branch 'main' into fix-reasoning-effort-optional

enyst commented:
**Hello, I’m OpenHands-GPT-5.** (posting here since I cannot reply directly under the code thread with this token)

Reply intended for: https://github.com/OpenHands/software-agent-sdk/pull/1004/files#r2491878129

- **Example used:** `examples/01_standalone_sdk/23_responses_reasoning.py`
- **Settings applied:** `reasoning_summary="auto"`, `reasoning_effort="medium"`

- **Proxy (LiteLLM eval host):**
  - Model: `litellm_proxy/gpt-5-mini-2025-08-07`
  - Base URL: `https://llm-proxy.eval.all-hands.dev`
  - Auth: `LITELLM_API_KEY`
  - Result: Success — agent completed the task; reasoning active; no errors.

- **Direct OpenAI:**
  - Model: `gpt-5-mini-2025-08-07`
  - Base URL: none
  - Auth: `OPENAI_API_KEY`
  - Result: 400 Bad Request — "Your organization must be verified to generate reasoning summaries." (param: `reasoning.summary`)


**Conclusion:**
- Through the proxy, "auto" works as expected.
- Direct OpenAI (unverified org): explicitly setting `reasoning_summary="auto"` triggers a 400. Keeping the default `None` avoids surprise failures; enabling summaries requires a verified org.

Chesars committed:
Set reasoning_effort default to high

Chesars committed:
Merge branch 'fix-reasoning-effort-optional' of https://github.com/Chesars/software-agent-sdk into fix-reasoning-effort-optional

Chesars committed:
Merge branch 'main' into fix-reasoning-effort-optional

xingyaoww reviewed (approved):
LGTM thanks!

Chesars commented:
The SWE-Bench team recently released a new Benchmark called CodeClash. Sounds interesting

xingyaoww commented:
@OpenHands please make sure pre-commit run --all-files passes

openhands-agent committed:
Apply ruff formatting to test_llm_config.py

Co-authored-by: openhands <openhands@all-hands.dev>
